app/src/main/java/io/appium/settings/location/LocationBuilder.java

```diff
@@ -0,0 +1,83 @@
+/*
+  Copyright 2012-present Appium Committers
+  <p>
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  <p>
+  http://www.apache.org/licenses/LICENSE-2.0
+  <p>
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+package io.appium.settings.location;
+
+import android.location.Location;
+import android.os.Build;
+import android.os.SystemClock;
+
+
+public class LocationBuilder {
+
+    final private String providerName;
+
+    private double latitude = 0.0;
```

Comment:
I think we should rather assign null to these values and only set them if they are really provided. Not sure if it is safe to assume `speed` and `bearing` and `accuracy` are zeroes if they are not provided

Reply:
from the source code: `android/location/Location.java`

we can see it's safe to set the default value to 0.

Reply:
You're correct, we should not set the default value to 0 for `speed` `bearing` `accuracy`